src/components/playlist-modal/PlaylistModal.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { Fragment, useState, useRef } from 'react';
+import ReactDom from 'react-dom';
+import './PlaylistModal.css';
+import { usePlaylists } from '../../hooks';
+import { textFormatter } from '../../utils';
+import { v4 as uuid } from 'uuid';
+
+const Backdrop = props => {
+  return <div onClick={() => props.onHide()} className="backdrop"></div>;
+};
+
+const overlay = document.getElementById('overlay');
+
+const PlaylistModal = props => {
+  const [wantNewPlaylist, setWantNewPlaylist] = useState(false);
+  const { playlists, createNewPlaylist, addToPlaylist, removeFromPlaylist } =
+    usePlaylists();
+
+  const newPlaylistRef = useRef();
+
+  const createNewPlaylistHandler = async () => {
+    const newPlaylist = newPlaylistRef.current.value.toLowerCase();
+
+    const error = await createNewPlaylist({
+      title: newPlaylist,
+      videos: [props.video],
+    });
+
+    if (!error) setWantNewPlaylist(false);
+  };
+
+  const playlistHandler = e => {
+    e.target.checked
+      ? addToPlaylist({ playlistTitle: e.target.value, video: props.video })
+      : removeFromPlaylist({
+          playlistTitle: e.target.value,
+          videoId: props.video._id,
+        });
+  };
+
+  return (
+    <Fragment>
+      {ReactDom.createPortal(<Backdrop onHide={props.onHide} />, overlay)}
+      {ReactDom.createPortal(
+        <div className="playlist-modal">
+          <div className="flex align-center space-between">
+            <h4>Save to...</h4>
+            <button className="btn-modal-dismiss" onClick={props.onHide}>
+              <i class="fas fa-times"></i>
+            </button>
+          </div>
+          <div className="hr-line solid thin"></div>
+          <ul>
+            {/* <li>
+              <input type="checkbox" value="watch later" id="watch-later" />
+              <label htmlFor="watch-later">Watch Later</label>
+            </li> */}
+
+            {playlists.map(el => (
```

Comment:
Suggestion: You can write something meaningful name instead of "el" while you use map.